src/soko_loader.rs

```diff
@@ -179,6 +181,96 @@ fn parse_sokoban_level(tokens: &[Token]) -> Result<World, String> {
     }
 }
 
+/// In the level format we cant tell what tiles are floors and what tiles are empty
+/// This function takes a board, and working from the outside of the board in culls
+/// any floor tiles with an empty neighbour.
+pub fn cull_outer_tiles(board: &mut Array2<Tile>) -> &Array2<Tile> {
+    let (height, width) = board.dim();
+    // Check first and last column for Floor tiles to cull
+    for yi in 0..height {
+        if matches!(board[[yi, 0]], Tile::Floor) {
+            cull_tiles([yi, 0], board);
+        }
+        if matches!(board[[yi, width - 1]], Tile::Floor) {
+            cull_tiles([yi, width - 1], board);
+        }
+    }
+    // Check first and last row for Floor tiles to cull
+    for xi in 0..width {
+        if matches!(board[[0, xi]], Tile::Floor) {
+            cull_tiles([0, xi], board);
+        }
+        if matches!(board[[height - 1, xi]], Tile::Floor) {
+            cull_tiles([height - 1, xi], board);
+        }
+    }
+    board
+}
+
+pub fn cull_tiles(index: [usize; 2], board: &mut Array2<Tile>) -> &Array2<Tile> {
+    // A guard so we don't cull any non-floor tiles
+    if !matches!(board[index], Tile::Floor) {
+        return board;
+    }
+    let (height, width) = board.dim();
+    let [yi, xi] = index;
+    let last_col = width - 1;
+    let last_row = height - 1;
+
+    // We'll use wrapping subration here because the adjacent_indexes slice further down is set up
+    // to not include the wrapping indexes.
+    let directions = [
+        [yi + 1, xi],
+        [yi.wrapping_sub(1), xi],
+        [yi, xi + 1],
+        [yi, xi.wrapping_sub(1)],
+    ];
+    let [bottom, top, right, left] = directions;
+
+    // Figure out what directions we need to check for adjacent emptys
+    //
+    // We could do something programatic here, but this code should never change so I feel that
+    // iterating the edgecases is more clear to the reader than writing some loop to conditionally
+    // build this slice.
```

Comment:
@ColtonPhillips What do you think of this? I think it's pretty clear but maybe iterating and having dynamic bounds check is better than iterating all the edge cases.

If this makes sense without thinking too hard I'll keep it, otherwise I'll do something else.